src/platform/pantherlake/include/platform/drivers/alh.h

```diff
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ */
+
+#ifdef __SOF_DRIVERS_ALH__
+
+#ifndef __PLATFORM_DRIVERS_ALH__
+#define __PLATFORM_DRIVERS_ALH__
+
+#include <stdint.h>
+
+#endif /* __PLATFORM_DRIVERS_ALH__ */
+
+#else
+
+#error "This file shouldn't be included from outside of sof/drivers/alh.h"
```

Comment:
`#error` is fatal so there's no need to
- enclose the entire file in this conditional and increase its nesting level
- separate the condition and the corresponding #error by wedging the entire file between them

  https://www.google.com/search?q=early+return+pattern
```
#ifndef __SOF_DRIVERS_ALH__
#error "This file shouldn't be included from outside of sof/drivers/alh.h"
#endif

/* normal activity can resume */

```